include/tins/sniffer.h

```diff
@@ -573,13 +577,44 @@ namespace Tins {
         unsigned _timeout;
     };
 
+    #if TINS_IS_CXX11
+
+    template<bool B, class T = void>
```

Comment:
This looks good. Could you move this structs/helper functions to internals.h, inside the Internals namespace? It's the only thing that I'd change.